pandas/_libs/parsers.pyx

```diff
@@ -118,6 +118,8 @@ cdef:
     float64_t NEGINF = -INF
     int64_t DEFAULT_CHUNKSIZE = 256 * 1024
 
+DEFAULT_BUFFER_HEURISTIC = 2 ** 20
```

Comment:
Can this be set as a property of the TextReader? It is a bit ambiguous in a pyx file but in the global namespace all caps I would expect this to be a compile time constant; attaching it as a property makes things clearer

Reply:
Or if this is just for testing maybe you can patch `buffer_lines` directly? The naming here is a bit unclear when scoped outside of the initializer

Reply:
Do you have the same concern about `DEFAULT_CHUNKSIZE` above too?

Ideally I think these magic numbers should at least be made obvious so they can be configured or removed https://github.com/pandas-dev/pandas/issues/53781 and it being in the TextReader would make that less obvious?

Reply:
Unless I am mistaken with how Cython generates this comparing this to `DEFAULT_CHUNKSIZE` is exactly the problem; that is a compile time constant whereas this value can be modified at runtime, but they both look the same

Reply:
Hmm does declaring it `cpdef` help at all? Not worth going down a rabbit hole over if its a hold up

Reply:
Here are my unsuccessful attempts so far just to double check

```
    cdef public:
        int64_t leading_cols, table_width, DEFAULT_BUFFER_HEURISTIC=2**20
                                          ^
------------------------------------------------------------

pandas/_libs/parsers.pyx:365:43: Cannot assign default value to fields in cdef classes, structs or unions
```

```
    cpdef DEFAULT_BUFFER_HEURISTIC=2**20
                                 ^
------------------------------------------------------------

pandas/_libs/parsers.pyx:377:34: Cannot assign default value to fields in cdef classes, structs or unions
```

Reply:
Cool thanks for checking. Let's not get hung up on it for now then - I think just a wart between the C/Python and how they get expressed in the Cython global namespace. Can always come back and refactor if we establish a better pattern

Reply:
Thanks for confirming!